File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/TA/Hierarchy/Project.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System.Collections;
using Db4objects.Db4o.Activation;
using Db4objects.Db4o.Tests.Common.TA;
using Db4objects.Db4o.Tests.Common.TA.Collections;
using Db4objects.Db4o.Tests.Common.TA.Hierarchy;

namespace Db4objects.Db4o.Tests.Common.TA.Hierarchy
{
	internal class Project : ActivatableImpl
	{
		internal IList _subProjects = new PagedList();

		internal IList _workLog = new PagedList();

		internal string _name;

		public Project(string name)
		{
			_name = name;
		}

		public virtual void LogWorkDone(UnitOfWork work)
		{
			// TA BEGIN
			Activate(ActivationPurpose.Read);
			// TA END
			_workLog.Add(work);
		}

		public virtual long TotalTimeSpent()
		{
			// TA BEGIN
			Activate(ActivationPurpose.Read);
			// TA END
			long total = 0;
			IEnumerator i = _workLog.GetEnumerator();
			while (i.MoveNext())
			{
				UnitOfWork item = (UnitOfWork)i.Current;
				total += item.TimeSpent();
			}
			return total;
		}
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/common/ta/hierarchy/Project.java
/* Copyright (C) 2007   Versant Inc.   http://www.db4o.com */
package com.db4o.db4ounit.common.ta.hierarchy;import java.util.*;

import com.db4o.activation.*;
import com.db4o.db4ounit.common.ta.*;

/**
 */@decaf.Ignore(decaf.Platform.JDK11)
class Project extends ActivatableImpl {		List _subProjects = new com.db4o.db4ounit.common.ta.collections.PagedList();		List _workLog = new com.db4o.db4ounit.common.ta.collections.PagedList();		String _name;		public Project(String name) {		_name = name;	}		public void logWorkDone(UnitOfWork work) {		// TA BEGIN		activate(ActivationPurpose.READ);		// TA END		_workLog.add(work);	}	public long totalTimeSpent() {				// TA BEGIN		activate(ActivationPurpose.READ);		// TA END				long total = 0;		Iterator i = _workLog.iterator();		while (i.hasNext()) {			UnitOfWork item = (UnitOfWork) i.next();			total += item.timeSpent();		}		return total;	}}